toolkit/themes/shared/design-system/config.js

```diff
@@ -82,64 +100,150 @@ ${formattedVars}
+ * 
+ * @param {string} surface - Desktop surface, either "brand", "platform", or "shared".
+ * @returns {function}
+ *  Matcher function for determining if a token's value needs to undergo
+ *  a light-dark transform.
+ */
+function getMatcher(surface) {
+  if (surface != "shared") {
+    return token =>
+      token.original.value[surface]?.light &&
+      token.original.value[surface]?.dark;
+  }
+  return token => token.original.value.light && token.original.value.dark;
+}
+
+/**
+ * Determines the correct transformer function to use for a given surface.
+ *
+ * @param {string} surface - Desktop surface, either "brand", "platform", or
+ * "shared".
+ * @returns {function} 
+ *  Transform function that uses the token's original value to create a new
+ *  "default" light-dark value and updates the original value object.
+ */
+function getTransformer(surface) {
+  if (surface != "shared") {
+    return token => {
+      let lightDarkVal = `light-dark(${token.original.value[surface].light}, ${token.original.value[surface].dark})`;
+      token.original.value[surface].default = lightDarkVal;
+      return token.value;
+    };
+  }
+  return token => {
+    let value = `light-dark(${token.original.value.light}, ${token.original.value.dark})`;
+    token.original.value.default = value;
+    return value;
+  };
 }
 
 module.exports = {
   source: ["design-tokens.json"],
-  transform: {
-    defaultTransform: {
-      type: "value",
-      transitive: true,
-      name: "defaultTransform",
-      matcher: token => token.original.value.default,
-      transformer: token => token.original.value.default,
-    },
-    lightDarkTransform: {
-      type: "value",
-      transitive: true,
-      name: "lightDarkTransform",
-      matcher: token => token.original.value.light && token.original.value.dark,
-      transformer: token => {
-        return `light-dark(${token.original.value.light}, ${token.original.value.dark})`;
-      },
-    },
-  },
   format: {
-    "css/variables/hcm": hcmFormatter,
+    "css/variables/shared": createDesktopFormat(),
+    "css/variables/brand": createDesktopFormat("brand"),
+    "css/variables/platform": createDesktopFormat("platform"),
   },
   platforms: {
     css: {
-      // The ordering of transforms matter, so if we encountered
-      // "light", "dark", and "default" in the value object then
-      // this ordering would ensure that the "default" value is
-      // used to generate the token's value.
       transforms: [
         ...StyleDictionary.transformGroup.css,
-        "lightDarkTransform",
-        "defaultTransform",
+        ...["shared", "platform", "brand"].map(createLightDarkTransform),
       ],
       buildPath: "build/css/",
       files: [
         {
           destination: "tokens-shared.css",
-          format: "css/variables/hcm",
+          format: "css/variables/shared",
+          options: {
+            outputReferences: true,
+            showFileHeader: false,
+          },
```

Comment:
I believe we can move this `options` object up to the CSS platform level so that we don't need to specify it for each generated file. If you want to do that as part of this, sweet. If not, I went ahead and filed https://mozilla-hub.atlassian.net/browse/FIDEFE-4673 to handle this